core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -41,6 +43,15 @@ abstract class SparkFunSuite
     }
   }
 
+  // helper function
+  protected final def getFile(file: String): File = {
```

Comment:
@srowen getTestResourceFile and getTestResourcePath look better. Thanks.

URL class doesn't have a method like getCanonicalFile. It has getFile only.

Also, I tested  Paths.get(... toURI).toFile. The only difference I noticed is that it keeps spaces as usual, but getFile(file).getCanonicalPath converts spaces to "%20". I suppose they are both OK.